lib/internal/modules/esm/import_map.js

```diff
@@ -0,0 +1,173 @@
+'use strict';
+const { isURL, URL } = require('internal/url');
+const {
+  ObjectEntries,
+  ObjectKeys,
+  SafeMap,
+  ArrayIsArray,
+  StringPrototypeStartsWith,
+  StringPrototypeEndsWith,
+  StringPrototypeSlice,
+  ArrayPrototypeReverse,
+  ArrayPrototypeSort,
+} = primordials;
+const { codes: { ERR_INVALID_IMPORT_MAP } } = require('internal/errors');
+const { shouldBeTreatedAsRelativeOrAbsolutePath } = require('internal/modules/helpers');
+
+class ImportMap {
+  #baseURL;
+  #imports = new SafeMap();
+  #scopes = new SafeMap();
+  #specifiers = new SafeMap()
+
+  constructor(raw, baseURL) {
+    this.#baseURL = baseURL;
+    this.process(raw, this.#baseURL);
```

Comment:
process function doesn't take a second parameter?

Reply:
Ah, this is from before I moved it onto the class. 👍